cibuildwheel/__main__.py

```diff
@@ -160,7 +160,7 @@ def main_inner(global_options: GlobalOptions) -> None:
     parser.add_argument(
         "package_dir",
         metavar="PACKAGE",
-        default=Path("."),
+        default=Path(),
```

Comment:
`Path()` is less self-documenting, could add a comment or revert?

```suggestion
        default=Path(),  # working directory
```

```suggestion
        default=Path("."),
```

Reply:
or `Path.cwd()`

Reply:
I wondered about that, but `Path.cwd()` would be an absolute path, which might make some logging more verbose unnecessarily.

Reply:
Not a big deal fyi @henryiii, I can learn to like it I guess, was just curious if anybody agreed with me.

Reply:
The Ruff recommendation is to use the default constructor `Path()` instead of `Path(".")` or `Path("")`. (Yes, `Path.cwd()` is different). I'm happy with either as long as it's consistent, and there's a check to enforce `Path()`. IMO, it's pretty clear, not sure what else `Path()` could be. (And, in pathlib, you can completely avoid `"."` and `".."`)

Reply:
If you're happy with either, and we're looking for a decision, my preference for readability, would be `Path(".")`.

> not sure what else `Path()` could be

Off the top of my head, it could be working dir, `Path(__file__)`, `Path(__file__).parent`, the filesystem root, the user home dir. It does seem weird to me that `Path() / 'bin'` and `Path('.') / 'bin'` are the same but `Path() / 'bin'` and `Path('/bin')` are different.

Reply:
I don't have a way to enforce that it's always `Path(".")`, but I do have a way to enforce that it's always `Path()` (https://docs.astral.sh/ruff/rules/path-constructor-current-directory), and I value consistency over either form. I guess I could manually do a search for the characters, since we enforce consistent spacing and quoting.

Almost all the forms you mention are absolute paths (`__file__` is always absolute starting in 3.9, 3.4-3.8 it could be relative for scripts), which are not the same as a relative path. (FWIW, Plumbum's path system didn't allow relative paths - making one was a string, which IMO was simpler)

https://docs.python.org/3/library/pathlib.html#pure-paths states "When pathsegments is empty, the current directory is assumed", so it's basically `"."` is default. They've designed it so you never have to use "." and "..", though you can. `Path() / "bin"` and `Path("bin")` are the same.

Reply:
FYI, there's also a rule for `Path(".").resolve()` -> `Path.cwd()`, I can apply that. That's almost every usage of `Path(".")` that we have - there are only two that are actually relative paths.

```
cibuildwheel/__main__.py:163:        default=Path("."),
cibuildwheel/macos.py:722:                        project=Path(".").resolve(),
cibuildwheel/macos.py:737:                        test_cwd = Path(".").resolve()
cibuildwheel/options.py:76:            package_dir=Path("."),
cibuildwheel/pyodide.py:295:            if str(Path(".").resolve()).startswith("/tmp"):
cibuildwheel/pyodide.py:296:                extra_mounts.append(str(Path(".").resolve()))
cibuildwheel/pyodide.py:416:                    project=Path(".").resolve(),
cibuildwheel/pyodide.py:430:                    test_cwd = Path(".").resolve()
cibuildwheel/windows.py:552:                    project=Path(".").resolve(),
cibuildwheel/windows.py:566:                    test_cwd = Path(".").resolve()
```

Also, `cibuildwheel/pyodide.py:295: if str(Path(".").resolve()).startswith("/tmp"):` will trigger on `/tmp*`, I wonder if that was intentional?

Reply:
Okay, let's go with `Path()`.

> FYI, there's also a rule for `Path().resolve()` -> `Path.cwd()`, I can apply that.

Sounds reasonable.

> Also, `cibuildwheel/pyodide.py:295: if str(Path(".").resolve()).startswith("/tmp"):` will trigger on `/tmp*`, I wonder if that was intentional?

Unlikely, based on context. I think that could be changed to `if str(Path.cwd()).startswith("/tmp/"):`

Reply:
Sounds good to me.